geolocator_android/android/src/main/java/com/baseflow/geolocator/LocationServiceHandlerImpl.java

```diff
@@ -42,7 +42,14 @@ void stopListening(){
         channel = null;
     }
 
-    void setActivity(@Nullable Activity activity) { this.activity = activity;}
+    void setActivity(@Nullable Activity activity) {
+        if (this.activity == activity) return;
```

Comment:
Please add braces around the body of the `if` statement. With our plugins we try to adhere to the same [conventions](https://google.github.io/styleguide/javaguide.html#s4.1-braces) used by the first-party Flutter plugins.
```suggestion
        if (this.activity == activity) {
          return;
        }
```